src/ChannelImpl.cpp

```diff
@@ -65,9 +66,7 @@ void ChannelImpl::Push(PacketBase* packet) {
   std::lock_guard<AdaptiveSpinLock> lock(GetQueuedMutex());
   assert((int)queued_packets_.size() < capacity_);
   Timer::Time earliest = queued_packets_.empty() ? Timer::kMaxTime : PeekNext();
-  queued_packets_.push_back(packet);
-  std::push_heap(queued_packets_.begin(), queued_packets_.end(),
-                 PacketOrdering());
+  queued_packets_.push_heap(packet);
```

Comment:
Thanks to skipping the constant instantiation of PacketOrdering() this version can even be faster (if the compiler did not optimize out those empty classes). Super.

Reply:
As far as I know, even empty classes will not be optimized out and will occupy a single byte in memory to guarantee that two distinct objects of the same type will have different addresses. I'm not sure if there was any serious overhead to instantiating it always, because it just boiled down to a simple allocation. Better to allocate something only once, though!